src/app/dataset-view/additional-components/lineage-component/lineage.component.ts

```diff
@@ -15,125 +14,31 @@ import { LineageGraphNodeKind, LineageGraphNodeData, LineageNodeAccess } from ".
 export class LineageComponent extends BaseComponent implements OnInit {
     @Output() onClickNodeEmit = new EventEmitter<Node>();
 
-    public lineageGraphLink: Edge[] = [];
-    public lineageGraphNodes: Node[] = [];
-    public isAvailableLineageGraph = false;
+    public lineageGraphLinks$: Observable<Edge[]>;
+    public lineageGraphNodes$: Observable<Node[]>;
     public currentDataset: DatasetLineageBasicsFragment;
+    public currentDataset$: Observable<DatasetLineageBasicsFragment>;
```

Comment:
Both variable and observable? Very weird. In similar cases, when introducing Rx-style I've organized subscriptions to avoid creating another state variable.

Reply:
I double checked. You are no longer using the `currentDataset` anywhere, only the observable version.

Reply:
Removed